L1Trigger/L1THGCal/src/concentrator/HGCalConcentratorSuperTriggerCellImpl.cc

```diff
@@ -54,13 +54,19 @@ void HGCalConcentratorSuperTriggerCellImpl::createAllTriggerCells(
         continue;
       }
 
-      //To guard against the case in v8 geometry where 
-      //there might be different thicknesses within a module 
-      if ( superTCmapping_.getCoarseTriggerCellId(id)!=s.second.getSTCId() ){
-	continue;
+      DetId tc_Id(id);
+      if (tc_Id.det() == DetId::Forward) {  //V8
```

Comment:
The check should be on V8/V9 geometry, not V8/V9 detids. Because V8 detids can be used with the V9 geometry, but there shouldn't be thickness mix in that case. So the `isV9Geometry()` method of the trigger geometry should be used.